src/LiveComponent/src/Hydration/AbstractDoctrineHydrationExtension.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Hydration;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ManagerRegistry;
+use Doctrine\Persistence\ObjectManager;
+
+/**
+ * @author Jean-Paul van der Wegen <info@jpvdw.nl>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+abstract class AbstractDoctrineHydrationExtension
+{
+    /**
+     * @param ManagerRegistry[] $managerRegistries
+     */
+    public function __construct(
+        private iterable $managerRegistries,
+    ) {
+    }
+
+    protected function objectManagerFor(string $class): ?ObjectManager
+    {
+        if (!class_exists($class)) {
+            return null;
+        }
+
+        // todo cache/warmup an array of classes that are "doctrine objects"
+        foreach ($this->managerRegistries as $registry) {
+            if ($om = $registry->getManagerForClass($class)) {
+                return self::ensureManagedObject($om, $class);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Ensure the $class is not embedded or a mapped superclass.
+     */
+    private static function ensureManagedObject(ObjectManager $om, string $class): ?ObjectManager
+    {
+        if (!$om instanceof EntityManagerInterface) {
+            // todo might need to add some checks once ODM support is added
+            return $om;
+        }
+
+        $metadata = $om->getClassMetadata($class);
+
+        if ($metadata->isEmbeddedClass || $metadata->isMappedSuperclass) {
+            return null;
+        }
+
+        return $om;
+    }
+
+    /**
+     * @psalm-param class-string<T> $className
```

Comment:
Is `@param class-string<T>` possible ?

Reply:
Yes it is, i changed to non-psalm version.